firefoxThemes/waifus/tab.js

```diff
@@ -24,16 +86,60 @@ function keyConfirm(e) {
   }
 }
 
+function setThemedFavicon(currentTheme) {
+  const faviconOptions = {width: 32, height: 32};
+  const faviconSVG = buildSVG(currentTheme, faviconOptions)
+  svgToPng(faviconSVG, faviconOptions, (imgData) => {
+    let link = document.querySelector("link[rel~='icon']");
+    if (!link) {
+      link = document.createElement('link');
+      link.rel = 'icon';
+      document.getElementsByTagName('head')[0].appendChild(link);
+    }
+    link.href = imgData;
+  });
+}
+
+function setThemedSearchInputIcon(currentTheme) {
+  const searchOptions = {width: 24, height: 24};
+  const searchSVG = buildSVG(currentTheme, searchOptions)
+  svgToPng(searchSVG, searchOptions, (imgData) => {
+    const pngImage = document.createElement('img');
+    pngImage.src = imgData;
+    const style = `input { background: white url(${pngImage.src}) 12px center no-repeat; }`
+    const styleTag = document.createElement("style");
+    styleTag.append(style);
+    document.head.append(styleTag);
+  });
+}
+
+function setThemedAboutIcon(currentTheme) {
+  const aboutOptions = {width: 96, height: 96};
+  const aboutSVG = buildSVG(currentTheme, aboutOptions)
+  svgToPng(aboutSVG, aboutOptions, (imgData) => {
+    const logo = document.querySelector("div[class='logo']")
+    const pngImage = document.createElement('img');
+    logo.appendChild(pngImage);
+    pngImage.src = imgData;
+  });
+}
+
 function applyTabListeners() {
-  browser.storage.local.get(["showWidget"])
+  browser.storage.local.get(["showWidget", "waifuThemes", "currentThemeId"])
     .then((storage) => {
+      const currentTheme = storage.waifuThemes.themes[storage.currentThemeId] ||
+        storage.waifuThemes.themes["19b65ec8-133c-4655-a77b-13623d8e97d3"];
+      const root = document.querySelector(':root');
+      root.style.setProperty('--accent-color', currentTheme.definition.colors.accentColor);
+      root.style.setProperty('--base-background-color', currentTheme.definition.colors.baseBackground);
+
       if (storage.showWidget === undefined || storage.showWidget) {
         document.querySelector("body").innerHTML =
           `
     <main>
     <div class="logo-and-wordmark">
         <div class="logo"></div>
-        <div class="wordmark"></div>
+        <div class="wordmark">Doki Theme for Firefox</div>
```

Comment:
The title for this on the homepage is quite long. It would be better to reduce the title to simply **Doki Theme** as we already know the platform this extension is used for.